### ISSUE TITLE
App crash when run some queries.

### ISSUE THREAD

**Issue by huyphams** [ClickHouse/clickhouse-cpp#266]
The lib would crash when we run the following queries:
1. Crash when trying to run a query with invalid quotes.
```
select * from ‟food‟;
```

2. Crash when trying to show some binary column 

Create table
```
create table tableplus_crash_example (col AggregateFunction(argMax, Int32, DateTime(3))) engine = Memory;
insert into tableplus_crash_example values (unhex('010000000001089170A883010000'));
```

Then run this query.
```
select col from tableplus_crash_example;
```

**Comment by Enmk** [comment]
Hi @huyphams thanks for your report, what is the server version?

**Comment by huyphams** [comment]
I can reproduce with the latest commit https://github.com/ClickHouse/clickhouse-cpp/commit/4774da1462addf64a270343523cf8f4e4609f159

**Comment by Enmk** [comment]
Ok, what is the **server** version? There was recently a breaking change on aggregated functions state serializations (I believe between 22.8 and 22.10) so I'm curious is that related...

**Comment by huyphams** [comment]
I'm testing with the ClickHouse version `21.12.3.32`

**Comment by huyphams** [comment]
Hi, @Enmk any update on this? we're holding back our release due to this issue. Should we move on or wait for the patch?
Thank you!

**Comment by Enmk** [comment]
@huyphams please see #273 for test sample, does that reproduces your case?

I tested that against 21.12 and it works fine:

**Comment by huyphams** [comment]
Hi @Enmk could you check again? I saw that all the tests failed. Btw can you also the first example as it would cause the app to crash?

**Comment by Enmk** [comment]
The crash itself is fixed here: #273

However, clickhouse-cpp currently doesn't support reading values of type `AggregateFunction(...)`, so you woudn't be able to SELECT the result. As a workaround, you might want to cast it to `String` or a `FixedString` in the SELECT statement.

**Comment by huyphams** [comment]
I'm still getting the crash with the query

```
select * from ‟untitled_table_5‟;
```
